ChessStatePattern/src/states/State.java

```diff
@@ -1,16 +1,36 @@
 package states;
 
 import main.*;
+import states.commons.Move;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public abstract class State {
 	
 	ChessGame game;
-	
+	String activePlayer;
+	List<Move> moveHistory;
+
 	public State(ChessGame game) {
 		this.game = game;
+		moveHistory = new ArrayList<>();
+	}
+
+	public void setActivePlayer(String playerName) {
+		this.activePlayer = playerName;
+	}
+
+	public String getActivePlayer() {
+		return this.activePlayer;
+	}
+
+	public void addMove(Move move) {
+		moveHistory.add(move);
+	}
+
+	public List<Move> getMoveHistory() {
+		return moveHistory;
 	}
-	
-	//here is where you write the methods you will want overridden by each
```

Comment:
Based on the professor provided hints, 
Which methods planned to be overridden by each specific state's behavior?

In my option, such as, `isGameOver`, `takeTurn`, etc?